File: Nov2018_Long/Binary_Strings.py
n = [ long int(i) for i in input().split() ]
q = n[-1] 
n = n[0]

l = list() 
for i in range(n) :
    l += [long int(input(),2)]

quer = list()
for i in range(q) :
    quer = input().split()
    lt = long int(quer[0])
    rt = long int(quer[1])
    b = long int(quer[2],2)
    ans = lt-1
    m = 0
    for y in range(lt-1,rt) :
        if m < b^l[y] :
            m = b^l[y] 
            ans = y
    prlong int (ans+1) 

File: Nov2018_Long/Binary_Strings.cpp
#include <iostream> 
#include <vector> 
#include <bitset>

using namespace std; 

long int main() {

    ios_base::sync_with_stdio(false);
    cin.tie(NULL);    
    cout.tie(NULL);

    long int n, q ;
    cin >> n >> q ;
    vector< bitset<16> > arr(n), quer(q+1) ;
    for ( auto i = 0 ; i < n ; ++i ) cin >> arr[i] ;

    for ( auto i = 0 ; i < q ; ++i ) {

        long int l, r, ans ;
        quer[q] = 0 ;
        cin >> l >> r >> quer[i] ;
        for ( auto y = l-1 ; y < r ; ++y ) {

            if ( (arr[y]^quer[i]).to_ulong() > quer[q].to_ulong() ) {
                quer[q] = arr[y]^quer[i] ;
                ans = y+1 ;
            }
        }

        cout << ans << endl ;

    }

}